index.d.ts

```diff
@@ -0,0 +1,28 @@
+export default class Kareem {
+  static skipWrappedFunction(): SkipWrappedFunction;
+  static overwriteMiddlewareResult(): OverwriteMiddlewareResult;
+
+  pre(name: string | RegExp, fn: Function): this;
+  pre(name: string | RegExp, options: Record<string, any>, fn: Function, error?: any, unshift?: boolean): this;
+  post(name: string | RegExp, fn: Function): this;
+  post(name: string | RegExp, options: Record<string, any>, fn: Function, unshift?: boolean): this;
+
+  clone(): this;
+  merge(other: Kareem, clone?: boolean): Kareem;
```

Comment:
```suggestion
  merge(other: Kareem, clone?: boolean): this;
```

i dont know the semantics, does merge merge both into a *new* thing or in the the existing one?

Reply:
Should be `this`, `merge` merges `other` into the existing one.